aten/src/ATen/cpu/vec/vec256/vec256_int.h

```diff
@@ -745,6 +745,259 @@ class Vectorized<int8_t> : public Vectorizedi {
   Vectorized<int8_t> le(const Vectorized<int8_t>& other) const;
 };
 
+template <>
+class Vectorized<uint8_t> : public Vectorizedi {
+private:
+  static const Vectorized<uint8_t> ones;
+public:
+  using value_type = uint8_t;
+  static constexpr int size() {
+    return 32;
+  }
+  using Vectorizedi::Vectorizedi;
+  Vectorized() {}
+  Vectorized(uint8_t v) { values = _mm256_set1_epi8(v); }
+  Vectorized(uint8_t val1, uint8_t val2, uint8_t val3, uint8_t val4,
+         uint8_t val5, uint8_t val6, uint8_t val7, uint8_t val8,
+         uint8_t val9, uint8_t val10, uint8_t val11, uint8_t val12,
+         uint8_t val13, uint8_t val14, uint8_t val15, uint8_t val16,
+         uint8_t val17, uint8_t val18, uint8_t val19, uint8_t val20,
+         uint8_t val21, uint8_t val22, uint8_t val23, uint8_t val24,
+         uint8_t val25, uint8_t val26, uint8_t val27, uint8_t val28,
+         uint8_t val29, uint8_t val30, uint8_t val31, uint8_t val32) {
+    values = _mm256_setr_epi8(val1, val2, val3, val4, val5, val6, val7, val8,
+                              val9, val10, val11, val12, val13, val14, val15, val16,
+                              val17, val18, val19, val20, val21, val22, val23, val24,
+                              val25, val26, val27, val28, val29, val30, val31, val32);
+  }
+  template <int64_t mask>
+  static Vectorized<uint8_t> blend(Vectorized<uint8_t> a, Vectorized<uint8_t> b) {
+    __at_align__ uint8_t tmp_values[size()];
+    a.store(tmp_values);
+    if (mask & 0x01)
```

Comment:
Why not doing this in a for loop with a pragma unroll?

Reply:
It seems the reason is that the second argument of _mm256_extract_epi64() and alike intrinsics has to be a constant value; namely, if attached patch applied, then compile errors pointing to this problem will be reported.
[patch.txt](https://github.com/pytorch/pytorch/files/10059709/patch.txt)

Reply:
Hmm, but [this](https://godbolt.org/z/WKvdPKrEs) works.

Reply:
I found mentioned [Godbolt](https://godbolt.org/z/WKvdPKrEs) works with `-O3`, but not with `-O2` that PyTorch uses.  Apparently, only `-O3` activates loop unrolling, and in this case compiler is able to generate required immediate argument for underlying assembly instruction.